modules/aws-http4s/src/smithy4s/aws/json/AwsSchemaVisitorJCodec.scala

```diff
@@ -105,6 +107,84 @@ private[aws] class AwsSchemaVisitorJCodec(cache: CompilationCache[JCodec])
     def decodeKey(in: JsonReader): Float = ???
 
     def encodeKey(x: Float, out: JsonWriter): Unit = ???
+  }
 
+  override def map[K, V](
+      shapeId: ShapeId,
+      hints: Hints,
+      key: Schema[K],
+      value: Schema[V]
+  ): JCodec[Map[K, V]] = {
+    value match {
+      case _: Schema.NullableSchema[_] => super.map(shapeId, hints, key, value)
+      case _ => flexibleNullParsingMap(shapeId, hints, key, value)
+    }
   }
+
+  // Overriding to be more flexible with null values : AWS
+  // expects clients to be able to process and discard null values
+  // in maps.
+  private def flexibleNullParsingMap[K, V](
+      shapeId: ShapeId,
+      hints: Hints,
+      sk: Schema[K],
+      sv: Schema[V]
+  ): JCodec[Map[K, V]] =
+    new JCodec[Map[K, V]] {
+      val expecting: String = "map"
+      val jk = self(sk)
+      val jv = self(sv)
+
+      override def canBeKey: Boolean = false
+
+      def decodeValue(cursor: Cursor, in: JsonReader): Map[K, V] =
+        if (in.isNextToken('{')) {
+          if (in.isNextToken('}')) Map.empty
+          else {
+            in.rollbackToken()
+            val builder = Map.newBuilder[K, V]
+            var i = 0
+            while ({
+              if (i >= maxArity) maxArityError(cursor)
+              val key = jk.decodeKey(in)
+              cursor.push(i)
+              if (in.isNextToken('n')) {
+                in.readNullOrError[Unit]((), "Expected null")
+              } else {
+                in.rollbackToken()
+                val value = cursor.decode(jv, in)
+                builder += (key -> value)
+              }
+              cursor.pop()
+
+              i += 1
+              in.isNextToken(',')
+            }) ()
+            if (in.isCurrentToken('}')) builder.result()
+            else in.objectEndOrCommaError()
+          }
+        } else in.decodeError("Expected JSON object")
+
+      def encodeValue(xs: Map[K, V], out: JsonWriter): Unit = {
+        out.writeObjectStart()
+        xs.foreach { kv =>
+          jk.encodeKey(kv._1, out)
+          jv.encodeValue(kv._2, out)
```

Comment:
do you have to encode the comma?

Reply:
I'm assuming we don't or we would have some tests blow up

Reply:
Yeah, `encodeKey` takes care of it